x/rollup/keeper/deposits.go

```diff
@@ -126,3 +195,60 @@ func (k *Keeper) mintETH(ctx sdk.Context, addr sdk.AccAddress, amount sdkmath.In
 
 	return &mintEvent, nil
 }
+
+// mintERC20 mints a bridged ERC-20 token to an account and returns the associated event.
+func (k *Keeper) mintERC20(
+	ctx sdk.Context, //nolint:gocritic // hugeParam
+	userAddr sdk.AccAddress,
+	erc20addr string,
+	amount sdkmath.Int,
+) (*sdk.Event, error) {
+	// use the "erc20/{l1erc20addr}" format for the coin denom
+	coin := sdk.NewCoin("erc20/"+erc20addr[2:], amount)
+	if err := k.bankkeeper.MintCoins(ctx, types.ModuleName, sdk.NewCoins(coin)); err != nil {
+		return nil, fmt.Errorf("failed to mint ERC-20 deposit coins to the rollup module: %v", err)
+	}
+	if err := k.bankkeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, userAddr, sdk.NewCoins(coin)); err != nil {
+		return nil, fmt.Errorf("failed to send ERC-20 deposit coins from rollup module to user account %v: %v", userAddr, err)
+	}
```

Comment:
Can we check what the standard Cosmos way to handle erc20s is? And document it here

Reply:
The only cosmos precedent I was able to find for erc20s is the [x/erc20](https://docs.evmos.org/protocol/modules/erc20#concepts) module from evmos. However, their token pairs need to be pre-registered with a EVM <-> Cosmos token mapping.

Our use case as an L2 makes this a novel problem for us if we want permissionless ERC-20 deposits without any whitelists or denom mapping. I believe that using the `erc20/{l1 address}` format for the cosmos token denom would be a functional and future-proof solution for us.

Please lmk if I misunderstood your question

Reply:
Nice, let's go with this.